### PR TITLE
Adding keep connection alive configuration

### PR DESCRIPTION
**Pull request by kaadhina**

### CODE REVIEW

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 1]
```diff
@@ -253,7 +253,8 @@ function Set-TestAgentConfiguration
         [String] $EnvironmentUrl,
         [String] $MachineName,
         [String] $Capabilities,
-        [System.Management.Automation.PSCredential] $AgentUserCredential
+        [System.Management.Automation.PSCredential] $AgentUserCredential,
+        [Bool] $keepConnectionAlive
```
Comment (nigurr) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: Keep it as optional field?

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 [hunk 2]
```diff
@@ -316,6 +317,16 @@ function Set-TestAgentConfiguration
     {
         $configArgs = $configArgs +  ("/Capabilities:{0}" -f $Capabilities)
     }
+    
+    if($keepConnectionAlive)
+    {
+        #To maintain compat with older test agents which will not have this parameter.
+        $consoleOptions = InvokeTestAgentConfigExe -Arguments "configureAsService /help" -Version $TestAgentVersion -UserCredential $MachineUserCredential    
+        if($consoleOptions.CommandOutput -match 'keepConnectionAlive')
+        {
```
Comment (nigurr) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: CommandOutput is string? or array?

**Comment by nigurr** [comment]
> function InvokeTestAgentConfigExe([string[]] $Arguments, [string] $Version, [System.Management.Automation.PSCredential] $UserCredential)

Didn't get the logic behind this

---
Refers to: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1:856 in b02ebd8. [](commit_id = b02ebd8da488d7581abda2e7010ab69d8cecaae5, deletion_comment = False)

**Comment by nigurr** [comment]
Approved

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 1 above)
Comment (kaadhina) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: This is taken care of below

File: Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1 (hunk 2 above)
Comment (kaadhina) [inline Tasks/DeployVisualStudioTestAgent/TestAgentConfiguration.ps1]: It is a string and -contains would not work. So leaving this as is.

**Pull request merged**